### PR TITLE
Friendly id for Profile URL 

### PR DESCRIPTION
**Pull request by AlanSoto31**
## Description

Add friendly id to profiles.

TODO:
Remove the code associated with UUIDs in the profile model.

## How has this been tested?

Please mark the tests that you ran to verify your changes. If difficult to test, consider providing instructions so reviewers can test.

- [x] Manual testing
- [ ] System tests
- [ ] Unit tests
- [ ] None

## Checklist

- [x] CI pipeline is passing
- [x] My code follows the conventions of this project
- [x] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (if applicable)
- [ ] I have added seed data to the database (if applicable)

## Release tasks

`rake profiles:set_slug`

## Screenshots/Loom
image

image

### CODE REVIEW

File: app/models/profile.rb [hunk 1]
```diff
@@ -40,6 +47,14 @@ def self.ransackable_attributes(_auth_object = nil)
     %w[name]
   end
 
+  def should_generate_new_friendly_id?
+    name_changed? || super
+  end
```
Comment (AlanSoto31) [inline app/models/profile.rb:52]: This method regenerates the slug when users change their profile name.

File: app/models/profile.rb [hunk 2]
```diff
@@ -40,6 +47,14 @@ def self.ransackable_attributes(_auth_object = nil)
...
+
+  def name_or_anonymous
+    name || "anonymous"
+  end
```
Comment (andresag4) [inline app/models/profile.rb:56]: When creating a Profile from Avo, the param name is set as empty string "", therefore the slug looks just as a uuid:
```
e106520e-3908-4730-ba93-c31a15627966
```

Creating from console works as expected as name is nil.
I would suggest using `.presence` to ensure the behaviour

```
  def name_or_anonymous
    name.presence || "anonymous"
  end
```

File: app/models/profile.rb (hunk 1 above)
Comment (andresag4) [reply app/models/profile.rb:52]: Lets also add this method in Session model, currently friendly_id stays the same even on title change

**Review by andresag4 — CHANGES_REQUESTED** [review]

**Comment by andresag4** [comment]
We could opt to just reduce the size of the uuid.
If we change SecureRandom.uuid to SecureRandom.hex(5) we get a much shorter value which is still very hard to duplicate


```Number of Possible Combinations: SecureRandom.hex(5) generates 1,099,511,627,776 possible combinations.
Probability of Duplicates: The probability of generating a duplicate increases with the number of generated values.
For 1,000 values, the probability of at least one duplicate is approximately 0.0000455%.
For 10,000 values, the probability of at least one duplicate is approximately 0.00455%.
For 100,000 values, the probability of at least one duplicate is approximately 0.455%.

3.3.3 :043 > SecureRandom.hex(5)
 => "4c167046fa" 
3.3.3 :044 > SecureRandom.hex(5)
 => "e3a9c880a2" 
3.3.3 :045 > SecureRandom.hex(5)
 => "847ab000d7" 
 ```

**Comment by LuigiR0jas** [comment]
The value proposition of this change decreases as its complexity increases. Given that it doesn't seem like a simple change, we're probably better off not implementing it in the first place

**Pull request closed without merging**